lib/trino-collect/src/main/java/io/trino/collect/cache/EvictableCache.java

```diff
@@ -104,8 +104,13 @@ public V get(K key, Callable<? extends V> valueLoader)
     {
         Token<K> newToken = new Token<>(key);
         Token<K> token = tokens.computeIfAbsent(key, ignored -> newToken);
+        Callable<? extends V> valueLoaderImpl = () -> {
+            // revive token if it got expired before reloading
+            tokens.computeIfAbsent(token.getKey(), ignored -> token);
```

Comment:
would this be equivalent?


```suggestion
            tokens.computeIfAbsent(key, ignored -> token);
```